### PR TITLE
Slightly improves the speed at which the crew monitor gets its data

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
Combines `scan` and `health_data` into one. Reducing proc overhead.
Also optimizes the proc a bit.

The real optimization can be done by removing the sortAssoc. It is the most brunt of the proc cost.
This can be done once we get TGUI. Since TGUI can sort things locally rather easily which saves server CPU time

stats (tested with 51 humans with vital scanners on):
```
Profile results (total time)
Proc Name                                                                                   Self CPU    Total CPU    Real Time     Overtime        Calls
---------------------------------------------------------------------------------------    ---------    ---------    ---------    ---------    ---------
OLD
/datum/repository/crew/proc/health_data                                                        0.121        0.295        0.295        0.000          400
/datum/repository/crew/proc/scan                                                               0.019        0.019        0.017        0.000           67

NEW
/datum/repository/crew/proc/health_data                                                        0.121        0.258        0.258        0.000          400
```

## Why It's Good For The Game
Faster speed is better

## Changelog
No CL since it's pure backend

### CODE REVIEW

**Comment by AffectedArc07** [comment]
https://github.com/ParadiseSS13/Paradise/pull/13683 May benefit this

**Comment by farie82** [comment]
> #13683 May benefit this

Yeeeeeeep.
We'll see which one gets merged faster. If that one is merged then I'll update this one

File: code/datums/cache/crew.dm [hunk 1]
```diff
@@ -18,50 +18,41 @@ GLOBAL_DATUM_INIT(crew_repository, /datum/repository/crew, new())
 	if(world.time < cache_entry.timestamp)
 		return cache_entry.data
 
-	var/tracked = scan()
-	for(var/obj/item/clothing/under/C in tracked)
+	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)
+		if(!istype(H.w_uniform, /obj/item/clothing/under))
```
Comment (Fox-McCloud) [inline code/datums/cache/crew.dm:22]: People can wear clothing uniforms?

File: code/datums/cache/crew.dm [hunk 2]
```diff
@@ -18,50 +18,41 @@ GLOBAL_DATUM_INIT(crew_repository, /datum/repository/crew, new())
 	if(world.time < cache_entry.timestamp)
 		return cache_entry.data
 
-	var/tracked = scan()
-	for(var/obj/item/clothing/under/C in tracked)
+	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)
```
Comment (Fox-McCloud) [inline code/datums/cache/crew.dm:21]: please update your PR and utilize the human list.

File: code/datums/cache/crew.dm (hunk 1 above)
Comment (farie82) [reply code/datums/cache/crew.dm:23]: What exactly do you mean?

Comment (farie82) [reply code/datums/cache/crew.dm]: Fixed

File: code/datums/cache/crew.dm (hunk 2 above)
Comment (farie82) [reply code/datums/cache/crew.dm]: Got it!

File: code/datums/cache/crew.dm [hunk 3]
```diff
@@ -18,50 +18,40 @@ GLOBAL_DATUM_INIT(crew_repository, /datum/repository/crew, new())
 	if(world.time < cache_entry.timestamp)
 		return cache_entry.data
 
-	var/tracked = scan()
-	for(var/obj/item/clothing/under/C in tracked)
+	for(var/dingy in GLOB.human_list)
```
Comment (Fox-McCloud) [inline code/datums/cache/crew.dm:21]: are people boats now? Are they dirty? *WHAT*

Comment (farie82) [reply code/datums/cache/crew.dm:21]: dirty? Yes. Ever seen a crewmember shower? I thought so

**Pull request merged**